src/main/java/seedu/address/model/Model.java

```diff
@@ -187,6 +207,26 @@ public interface Model {
      */
     boolean hasVideo(ReadOnlyLecture lecture, Video video);
 
+    /**
+     * Returns true if a video with the name {@code videoName} exists in {@code lecture}.
+     *
+     * @param lecture The lectre to check if t contains the video.
+     * @param videoName The video name to check if exist.
+     * @return True if a video with the name {@code videoName} exists in {@code lecture}. Otherwise, false.
+     */
+    boolean hasVideo(ReadOnlyLecture lecture, VideoName videoName);
+
+    /**
+     * Returns the video with {@code videoName} in the lecture {@code lecture}
+     * The lecture must exist in the tracker.
+     * The video must exist in {@code lecture}
+     *
+     * @param lecture The lecture that contains the video with {@code videoName}
+     * @param videoName Name of video to return
+     * @return Video with {@code videoName} found in {@code lecture}
+     */
+    Video getVideo(ReadOnlyLecture lecture, VideoName videoName);
```

Comment:
Don't think this is necessary, could you use `Model.getTracker().getModule(ModuleCode).getLecture(LectureName).getVideo(VideoName)`?